cmd/xl-v1-multipart.go

```diff
@@ -727,16 +722,12 @@ func (xl xlObjects) PutObjectPart(bucket, object, uploadID string, partID int, s
 	// Add the current part.
 	xlMeta.AddObjectPart(partID, partSuffix, newMD5Hex, size)
 
-	for index, disk := range onlineDisks {
-		if disk == nil {
-			continue
+	for i, disk := range onlineDisks {
+		if disk != OfflineDisk {
```

Comment:
Makes sense for multiple conditions in a loop:
```
for {
     if cond1 {
            continue
     }
     if cond2 {
            continue
     }
// logic....
}
```
But not for a single condition?!